prog.go

```diff
@@ -454,53 +454,11 @@ func (p *Program) MarshalBinary() ([]byte, error) {
 	return buf, nil
 }
 
-// Attach a Program to a container object fd
-func (p *Program) Attach(fd int, typ AttachType, flags AttachFlags) error {
```

Comment:
Instead of removing the complete function, could we keep the signature and use it as wrapper for `RawAttachProgram`? Same for `func (p *Program) Detach(fd int, typ AttachType, flags AttachFlags) error`. Maybe also add a comment that for newer kernel 5.7 link should be the prefer option.
As not everyone is already using 5.7 this might break a lot of stuff :/

Reply:
Yeah, that is a good point. We could document these as deprecated, and then remove these functions at a later time. In general I don't want to keep these around forever, so at some point we'll have to introduce breakage.

Reply:
Turns out this is impossible, since it creates an import cycle:

```
can't load package: import cycle not allowed
package github.com/cilium/ebpf
	imports github.com/cilium/ebpf/link
	imports github.com/cilium/ebpf
```

I don't think I can fix this.

Reply:
In that case (i.e. removing the functions) I'd strongly suggest to tag a release before this PR is merged.

Reply:
I've moved stuff around, and just kept the old functions pretty much as they are. This means we don't break the API anymore.